packages/api/src/external/carequality/command/cq-directory/create-or-update-cq-organization.ts

```diff
@@ -1,63 +1,104 @@
-import { errorToString } from "@metriport/shared/common/error";
+import { out } from "@metriport/core/util/log";
+import { capture } from "@metriport/core/util/notifications";
+import { errorToString } from "@metriport/shared";
 import { makeCarequalityManagementAPI } from "../../api";
 import { CQOrganization } from "../../organization";
 import { CQOrgDetails } from "../../shared";
+import { isCQDirectEnabledForCx } from "../../../aws/app-config";
 
 const cq = makeCarequalityManagementAPI();
 
-export async function createOrUpdateCQOrganization(orgDetails: CQOrgDetails): Promise<string> {
+export async function createOrUpdateCQOrganization(
+  cxId: string,
+  orgDetails: CQOrgDetails
+): Promise<string | undefined> {
   if (!cq) throw new Error("Carequality API not initialized");
   const org = CQOrganization.fromDetails(orgDetails);
-  const orgExists = await doesOrganizationExistInCQ(org.oid);
+  const orgExists = await doesOrganizationExistInCQ(cq, cxId, org.oid);
   if (orgExists) {
-    return updateCQOrganization(cq, org);
+    return updateCQOrganization(cq, cxId, org);
   }
-  return registerOrganization(cq, org);
+  return registerOrganization(cq, cxId, org);
 }
 
-export async function doesOrganizationExistInCQ(oid: string): Promise<boolean> {
-  if (!cq) throw new Error("Carequality API not initialized");
-  const org = await cq.listOrganizations({ count: 1, oid });
-  if (org.length > 0) {
-    return true;
+async function doesOrganizationExistInCQ(
+  cq: CarequalityManagementAPI,
+  cxId: string,
+  oid: string
+): Promise<boolean | undefined> {
+  const { log } = out(`CQ get (Organization) - CQ Org OID ${oid}`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    log(`CQ disabled for cx ${cxId}, skipping...`);
+    return undefined;
+  }
+  try {
+    const resp = await cq.listOrganizations({ count: 1, oid });
+    return resp.length > 0;
+  } catch (error) {
+    const msg = `Failure while getting Org @ CQ`;
+    log(`${msg}. Org OID: ${oid}. Cause: ${errorToString(error)}`);
+    capture.error(msg, {
+      extra: {
+        orgOid: oid,
+        context: `cq.org.get`,
+        error,
+      },
+    });
+    throw error;
   }
-  return false;
-}
-
-export async function getCqOrganization(oid: string): Promise<Organization | undefined> {
-  if (!cq) throw new Error("Carequality API not initialized");
-  const organizations = await cq.listOrganizations({ count: 1, oid });
-  return organizations[0];
 }
 
 async function updateCQOrganization(
   cq: CarequalityManagementAPI,
+  cxId: string,
   cqOrg: CQOrganization
-): Promise<string> {
-  console.log(`Updating organization in the CQ Directory with OID: ${cqOrg.oid}...`);
+): Promise<string | undefined> {
+  const { log } = out(`CQ update (Organization) - CQ Org OID ${cqOrg.oid}`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    log(`CQ disabled for cx ${cxId}, skipping...`);
+    return undefined;
+  }
```

Comment:
Idk if this check is necessary. This logic is only used by us to create and update orgs and `isCQDirectEnabledForCx` feels like its more so cxs dont make queries if they arent active yet

Reply:
No longer relevant. Removed.